File: lab1zad4.cpp
#include <iostream>
#include <string>
#include <vector>
#include <ctime>
#include <cstdlib>
using namespace std;

class Parrot {
private:
    vector<string> phrases;

public:
    Parrot(const string& initialPhrase) {
        phrases.push_back(initialPhrase);
    }

    void say(int repeatCount = 1) const {
        srand(time(0));
        string phrase = phrases[rand() % phrases.size()];
        for (int i = 0; i < repeatCount; ++i) {
            cout << phrase << endl;
        }
    }

    void addPhrase(const string& newPhrase) {
        phrases.push_back(newPhrase);
    }
};

int main() {
    Parrot myParrot("Hello!");
    myParrot.say();
    myParrot.addPhrase("I'm a parrot!");
    myParrot.say(3);
    return 0;
}

File: lab1zad4.java
import java.util.ArrayList;
import java.util.Random;

public class Parrot {
    private ArrayList<String> phrases;

    public Parrot(String initialPhrase) {
        phrases = new ArrayList<>();
        phrases.add(initialPhrase);
    }

    public void say(int repeatCount) {
        Random rand = new Random();
        String phrase = phrases.get(rand.nextInt(phrases.size()));
        for (int i = 0; i < repeatCount; i++) {
            System.out.println(phrase);
        }
    }

    public void addPhrase(String newPhrase) {
        phrases.add(newPhrase);
    }

    public static void main(String[] args) {
        Parrot myParrot = new Parrot("Hello!");
        myParrot.say(1);
        myParrot.addPhrase("I'm a parrot!");
        myParrot.say(3);
    }
}
